odl/solvers/advanced/proximal_operators.py

```diff
@@ -124,14 +133,184 @@ def cconj_prox_factory(step_size):
+                (IdentityOperator(y.space) - ConstantOperator(y)))
+
+    return translation_prox_factory
+
+
+def proximal_arg_scaling(prox_factory, scaling):
+    """Calculate the proximal of function F(x * scaling).
+
+    This is calculated according to the rule
+
+        prox[s * F( . * scaling)](x) =
+        1/scaling * prox[s * scaling^2 * F ](x * scaling)
+
+    where ``scaling`` is the scaling parameter, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    scaling : non-zero `float`
+        Scaling parameter
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [CP2011c]_.
+    """
+
+    scaling = float(scaling)
+
+    if scaling == 0:
+        return proximal_zero(prox_factory(1.0).domain)
+
+    def arg_scaling_prox_factory(step_size):
+        """Create proximal for the translation with a given step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * F( . * 1/a)`` where ``s`` is the
+            step size
+        """
+        prox = prox_factory(step_size * scaling**2)
+        return 1 / scaling * prox * scaling
+
+    return arg_scaling_prox_factory
+
+
+def proximal_quadratic_perturbation(prox_factory, a, u):
+    """Calculate the proximal of function F(x) + a * ||x||^2 + <u,x>.
+
+    This is calculated according to the rule
+
+        prox[s * (F( . ) + a * || . ||^2 + <u, . >)](x) =
+        c prox[s*f( . * c)]((x - s*u)*c)
+
+    where ``c`` is the constant c = 1/sqrt(s*2*a + 1), ``a`` is the scaling
+    parameter belonging to the quadratic term, ``u`` is the vector defining the
+    linear functional, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    a : non-negative `float`
+        Scaling of the quadratic term
+    u : Element in domain of F
+        Defines the linear functional
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [CP2011c]_. Note that this identity
+    is not the exact one given in the reference, but was recalculated for
+    arbitrary step lengths.
+    """
+
+    if a < 0:
+        raise ValueError('scaling parameter {} not non-negative'.format(a))
+
+    if not isinstance(u, LinearSpaceVector):
+        raise TypeError('vector {!r} not a LinearSpaceVector instance.'
+                        ''.format(u))
+
+    if not u.space == prox_factory(1).domain:
```

Comment:
We dont do this in the others and I'd skip it here. If we add it we'll add to all of them. This could possibly incur a cost, or even a crash (perhaps 1 is not a valid value?)